src/main/java/org/tugraz/sysds/runtime/data/TensorBlock.java

```diff
@@ -636,9 +673,46 @@ public CacheBlock slice(int rl, int ru, int cl, int cu, CacheBlock block) {
 		return null;
 	}
 
+	public TensorBlock slice(int[] offsets, TensorBlock block) {
+		assert offsets.length == block.getNumDims();
+		int[] srcIx = new int[offsets.length];
+		Array.copy(offsets, 0, srcIx, 0, offsets.length);
+		int[] destIx = new int[offsets.length];
+		for (int l = 0; l < block.getLength(); l++) {
+			block.set(destIx, get(srcIx));
+			int i = block.getNumDims() - 1;
```

Comment:
I added a TODO and will improve it later.

Reply:
Yesterday we talked about the `getNextIndex` procedure, I went ahead and compared it to using modulo for each dimension, this leads to about 12x worse performance for calculating the next index. Instead of 8.22ms we get 103.97ms for calculating all valid indexes for the dimensions `{8, 8, 8, 8, 8, 8, 8}`.